pkgs/development/tools/analysis/cppcheck/default.nix

```diff
@@ -42,6 +37,16 @@ stdenv.mkDerivation rec {
     cp cppcheck.1 $man/share/man/man1/cppcheck.1
   '';
 
+  doInstallCheck = true;
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    echo 'int main() {}' > ./installcheck.cpp
+    $out/bin/cppcheck ./installcheck.cpp > /dev/null
```

Comment:
This won't work when cross compiling, right?

Reply:
I believe it shouldn't be a problem, the tests shouldn't run when doing cross-compilation. See [6.5.9.1.1. doInstallCheck](https://nixos.org/manual/nixpkgs/stable/#var-stdenv-doInstallCheck):

> Cross compiled builds never run tests, no matter how doInstallCheck is set, as the newly-built program won’t run on the platform used to build it.